src/backend/mpi/swdsm.h

```diff
@@ -312,22 +310,45 @@ unsigned long getCacheIndex(unsigned long addr);
 /**
  * @brief Gives homenode for a given address
  * @param addr Address in the global address space
- * @param cloc Used to identify the call location in the code
  * @return Process ID of the node backing the memory containing addr
  */
-unsigned long getHomenode(unsigned long addr, char cloc = 0);
+argo::node_id_t get_homenode(std::size_t addr);
+/**
+ * @brief Gives homenode for a given address
+ * @param addr Address in the global address space
+ * @return Process ID of the node backing the memory containing addr
+ * @note This version does not invoke a first-touch call and instead
+ * returns -1 if an address has not been first-touched
+ */
+argo::node_id_t peek_homenode(std::size_t addr);
 /**
  * @brief Gets the offset of an address on the local nodes part of the global memory
  * @param addr Address in the global address space
- * @param cloc Used to identify the call location in the code
  * @return addr-(start address of local process part of global memory)
  */
-unsigned long getOffset(unsigned long addr, char cloc = 0);
+std::size_t get_offset(std::size_t addr);
+/**
+ * @brief Gets the offset of an address on the local nodes part of the global memory
+ * @param addr Address in the global address space
+ * @return addr-(start address of local process part of global memory)
+ * @note This version does not invoke a first-touch call and instead
+ * returns SIZE_MAX if an address has not been first-touched
```

Comment:
Is `SIZE_MAX` in the code clear enough, or should we have a constant (that is set to `SIZE_MAX`) and use that instead?

Reply:
The latest push should now detail that the return value is `::invalid_offset` for peek_offset (and `::invalid_node_id` for peek_homenode). These will link to the definitions in first_touch_allocation.hpp.